### PR TITLE
Enable write-implies-execute-never when applicable

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: core/arch/arm/kernel/generic_entry_a32.S [hunk 1]
```diff
@@ -189,6 +189,9 @@ UNWIND(	.cantunwind)
 	orr	r0, r0, #SCTLR_A
 	bic	r0, r0, #SCTLR_C
 	bic	r0, r0, #SCTLR_I
+#if defined(CFG_HWSUPP_MEM_PERM_WXN) && defined(CFG_CORE_RWDATA_NOEXEC)
```
Comment (jforissier) [inline core/arch/arm/kernel/generic_entry_a32.S]: Better to test only `CFG_CORE_RWDATA_NOEXEC` here.
Optionally, add a dependency check in a `core/arc/arm/arm.mk` (but see my other comment, I feel the HWSUPP config is useless):
```
ifeq ($(CFG_CORE_RWDATA_NOEXEC))
ifneq ($(CFG_HWSUPP_MEM_PERM_WXN),y)
$(error ...)
endif
endif
```

Comment (etienne-lms) [inline core/arch/arm/kernel/generic_entry_a32.S]: Some HW (C-A5, C-A9) do not support WXN and UWXN in SCTLR.
So we need this `CFG_HWSUPP_MEM_PERM_WXN`.
This is why the directive is not used in AARCH64, as all ARMv8-A do support SCTLR[WXN].

Comment (jforissier) [inline core/arch/arm/kernel/generic_entry_a32.S]: I understand that. But we can simply assume that platforms that don't support CFG_CORE_RWDATA_NOEXEC simply don't enable it, no?

Comment (etienne-lms) [inline core/arch/arm/kernel/generic_entry_a32.S]: No. At least for the few platforms lacking this hw hardening, op-tee core can still build a mmu that does rw-data-noexec. Weaker protection but at least optee legitimate mapping is clean.
The arm-tf does both: its mmu api never maps rwx memory and it sets STCRL[WXs], of course.

**Comment by etienne-lms** [comment]
rebased

File: core/arch/arm/kernel/generic_entry_a32.S (hunk 1 above)
Comment (jforissier) [inline core/arch/arm/kernel/generic_entry_a32.S]: Ah, now I get it. We really have two things to configure here. Sorry for the confusion.

**Comment by jforissier** [comment]
So, for both patches:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
Rebased, tag applied.

**Pull request merged**